addons/test_mail_full/tests/test_sms_composer.py

```diff
@@ -120,6 +120,21 @@ def test_composer_comment_field_w_template(self):
 
         self.assertSMSNotification([{'partner': self.test_record.customer_id, 'number': self.test_record.mobile_nbr}], 'Dear %s this is an SMS.' % self.test_record.display_name, messages)
 
+    def test_composer_default_recipient(self):
+        self.test_record.write({
+            'phone_nbr': '0123456789',
+        })
+        with self.with_user('employee'):
+            composer = self.env['sms.composer'].with_context(
+                    default_res_model='mail.test.sms', default_res_id=self.test_record.id,
+                ).create({
+                    'body': self._test_body,
+                    'number_field_name': 'phone_nbr',
+                })
+
+        self.assertFalse(composer.recipient_single_valid)
```

Comment:
I don't think it is useful to check if it is valid or not since your change does not impact that field. 🤔 

Let's see what the team in charged of those methods thinks about your changes. 🙂

Reply:
Indeed, but I guess the idea is to make sure we take the path that leads to the error 
(if it were valid, the 'partner' key would be Truthy, because _sms_get_default_partners() is Truthy for the purpose of the test). 

Personally, it doesn't bother me :)